lib/active_merchant/connection.rb

```diff
@@ -94,7 +94,12 @@ def request(method, body, headers = {})
 
     private
     def http
-      http = Net::HTTP.new(endpoint.host, endpoint.port, proxy_address, proxy_port)
+
+      http = if proxy_address == :DEFAULT
```

Comment:
If I'm reading [the docs and the source](https://ruby-doc.org/stdlib-2.4.1/libdoc/net/http/rdoc/Net/HTTP.html#method-c-new) right, can't we just have `@proxy_address = :ENV` by default and have Net::HTTP take care of it?

Reply:
That is my understanding as well.

Reply:
@bdewater Created https://github.com/activemerchant/active_merchant/pull/2800 with the revised approach.  Please review.